### ISSUE TITLE
Inject script doesn't work with auto-run feature

### ISSUE THREAD

**Issue by johnpryan** [dart-lang/dart-pad#2985]
There's a timing issue where using the `data-run` / `run=true` feature will correctly auto-run, but doesn't run the right code. Instead of running the injected code, it runs whatever `main.dart` was created by default.

DartPad probably needs to know that it is waiting on an injected script, and then wait for that code before triggering the auto-run feature.

cc: @parlough @tugorez

**Comment by tugorez** [comment]
This is awesome, thanks both @johnpryan and @parlough :D!

I was playing with `data-run="true"` and found that although it actually auto renders the UI it seems I'm still able to see the default program for a second. 

[dartpad_autorun.webm](https://github.com/dart-lang/dart-pad/assets/7821057/4f42d272-a618-4776-9090-68c97dcb65ea)

**Comment by parlough** [comment]
Yeah. I'm not sure how to avoid that 100% of the time without requiring a special flag to indicate not to run the initial script or gist. I might be able to adjust the order a bit and avoid running the initial script if a message was received from the embedder. I'll play around a bit.

**Comment by devoncarew** [comment]
@tugorez - I believe you're experimenting with re-mixing the dartpad UI a bit?

Instead of hosting the embedded view in a page, you might also consider forking and modifying the dartpad UI directly? As a Flutter web app, the UI components are reasonably modular, and it should be pretty quick to experiment with.
